syntax = "proto3";
package cal_counter;

message base_counter {
    int32 counter_id = 1;
    int64 value = 2;
}

message box_set
{
    repeated base_counter counter = 1;
    int32 box_id = 2;
    int32 node = 3;
}

message stream_elem
{
    int32 region_id_1 = 3;
    int32 region_id_2 = 4;
    int64 duration_us = 5;
    repeated box_set boxes = 6;
    int64 energy = 7;
    int32 core_frequncy = 8;
    int32 uncore_frequncy = 9;
    int32 region_id_1_event = 10;
    int32 region_id_2_event = 11;
}

message table_elem {
    string name = 2;
    int32 id = 3;
}

message region_elem {
    string name = 1;
    int32 id = 2;
}

message stream {
    repeated stream_elem elem = 1;
    repeated table_elem table = 2;
    repeated region_elem region = 3;
}